subsys/net/lib/lwm2m/Kconfig

```
@@ -190,6 +190,15 @@ config LWM2M_TLS_SESSION_CACHING
 	help
 	  Enabling this only when feature is supported in TLS library.
 
+config LWM2M_RD_CLIENT_AUTO_INIT
+	bool "Init RDClient automatically during device startup"
+	default y
+	help
+	  If this option is set, then the the RDClient is automatically
+	  initialized when the device is started. If you do not wish to do
+	  this, then disable this a call lwm2m_rd_client_init() in your
+	  application.
+
```

```suggestion
config LWM2M_RD_CLIENT_AUTO_INIT
	bool "Init RDClient automatically during device startup"
	default y
	help
	  If this option is set, then the the RDClient is automatically
	  initialized when the device is started. If you do not wish to do
	  this, then disable this a call lwm2m_rd_client_init() in your
	  application.

```

I propose that we don't add any Kconfigs just to test something.

---

Thanks for your review :) How would you solve this otherwise if you want to prevent rd_client_init from being called at SYS_INIT.

---

Why do you want to prevent `rd_client_init` being called at SYS_INIT?

---

If you need to set some custom fakes before the function is called, you can set your own `SYS_INIT` function in the unit test and set the custom fake there.

---

I think this PR is just fine, if you just remove this Kconfig.
There is no reason to prevent `rd_client_init()`. You have stubbed the `add_service` and in your unittests you set the stub and after that, call `rd_client_init()`, so the call will eventually call you own stub. 
So all seem to be working OK without preventing `SYS_INIT()` from happening.

---

Thanks. I committed as you proposed. @jarlamsa:I would like to keep `rd_client_init` to initialise rd client and in particular its internal state for every (sub) tests so that if one of the tests fails, every (sub) test that runs afterwards starts with an initial setup.

---

The problem is that the tests aren't isolated. So basically you are calling `init` for already initialized module and it doesn't test anything. I don't see any reason to omit the `SYS_INIT` in this case.